File: conformance/bundles/19-emphasis-parameter/MentionSteps.java
package dev.varar.conformance.bundle19;

import dev.varar.State;
import dev.varar.StepDefinitions;
import dev.varar.Steps;

/** Java sibling of {@code mention.steps.ts} (bundle {@code 19-emphasis-parameter}). */
public final class MentionSteps implements StepDefinitions<MentionSteps.Ctx> {

    record Ctx() implements State {}

    @Override
    public void register(Steps<Ctx> s) {
        s.state(Ctx::new);

        s.stimulus("I mention {emph}", (Ctx ctx, String who) -> ctx);
    }
}

File: conformance/bundles/19-emphasis-parameter/mention.steps.kt
@file:JvmName("MentionSteps")

// Kotlin sibling of mention.steps.ts / MentionSteps.java (bundle
// 19-emphasis-parameter).
package dev.varar.kotlin.conformance.bundle19

import dev.varar.kotlin.stimulus
import dev.varar.kotlin.steps

class Ctx

val steps = steps(::Ctx) {
    stimulus("I mention {emph}") { who: String -> this }
}
